eth/types/types.go

```diff
@@ -381,6 +382,10 @@ func NewTransaction(
 		result.BlobVersionedHashes = tx.BlobHashes()
 	}
 
+	if tx.Type() > types.BlobTxType {
```

Comment:
Can you explain this check here? And maybe add https://eip7702.io/#specification to the comments for reference.

Reply:
Good point, the checks can be a bit difficult to grasp. Added some helpful comments in https://github.com/onflow/flow-evm-gateway/pull/821/commits/3ceea86e0c7f51ed30ff31d6d1eff81c49f0ff9e .